app/javascript/controllers/request_item_controller.js

```diff
@@ -0,0 +1,32 @@
+import { Controller } from "@hotwired/stimulus";
+
+// Connects to data-controller="request-item"
+export default class extends Controller {
+  static targets = ["addButton", "addDest", "addTemplate"];
+
+  addItem() {
+    const template =
+      this.addTemplateTarget.content.cloneNode(true).firstElementChild;
+
+    const templateId = new Date().getTime();
+    const rendered = template.innerHTML.replace(
```

Comment:
This feels a lot like magic. Why are we replacing this particular regex, and what does the replacement signify?

Reply:
Yea, I was thinking about this afterward and I agree. Do you think it would be better to do something like 
```js
template.id = gsub(regex, template.id)
template.name = gsub(regex, template.id)
```
and maybe add a comment for what the regex does?

Reply:
Actually that does not work. The regex replaces multiple deeply nested values, I think the best that we could do would be to extract the replacement into another method and give it a nice name with a comment to explain the regex